smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/xml/StructDecodeXMLGenerator.kt

```diff
@@ -38,7 +38,7 @@ open class StructDecodeXMLGenerator(
             SmithyXMLTypes.Reader
```

Comment:
This additional condition fixes the error thrown when handling `EndEvent` that's returned as part of the event stream output of the call to `S3:SelectObjectContent`. The [S3 docs](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTSelectObjectAppendix.html) say the payload of this event is empty (not an empty XML object, just literally empty), which causes an issue with deserializing in SDK side. The additional condition checks if the struct being deserialized has any properties, and if it doesn't, returns a new instance rather than returning nil.